src/Admin/SiteHealth.php

```diff
@@ -76,11 +66,20 @@ public function __construct( MonitorCssTransientCaching $css_transient_caching,
 	 * Adds the filters.
 	 */
 	public function register() {
-		add_filter( 'site_status_tests', [ $this, 'add_tests' ] );
-		add_filter( 'debug_information', [ $this, 'add_debug_information' ] );
-		add_filter( 'site_status_test_result', [ $this, 'modify_test_result' ] );
-		add_filter( 'site_status_test_php_modules', [ $this, 'add_extensions' ] );
-		add_action( 'admin_print_styles-site-health.php', [ $this, 'add_styles' ] );
+
+		if ( is_admin() ) {
+			add_filter( 'site_status_tests', [ $this, 'add_tests' ] );
+			add_filter( 'debug_information', [ $this, 'add_debug_information' ] );
+			add_filter( 'site_status_test_result', [ $this, 'modify_test_result' ] );
+			add_filter( 'site_status_test_php_modules', [ $this, 'add_extensions' ] );
+			add_action( 'admin_print_styles-site-health.php', [ $this, 'add_styles' ] );
+		}
+
+		if ( wp_doing_ajax() ) {
+			add_action( 'wp_ajax_health-check-site-status', [ $this, 'ajax_site_status' ], 11 );
```

Comment:
This could instead run at the `wp_ajax_health-check-amp-page_cache` action, which would simplify the `ajax_site_status` method and could be renamed to `ajax_check_page_cache`, for example.

```suggestion
			add_action( 'ajax_site_status', [ $this, 'ajax_check_page_cache' ] );
```

Reply:
Hey @pierlon 

We can not change the action name, since it triggers from the "Health Check" plugin.

Reply:
Oh OK I see. However, this test is not running whenever the Health Check plugin is disabled though, as the AJAX action is now `health-check-amp-page_cache` instead of `wp_ajax_health-check-site-status`:

Health Check plugin disabled:

![image](https://user-images.githubusercontent.com/16200219/129707166-46595ea1-6a3f-4deb-aead-caa215afc10c.png)

Health Check plugin enabled:

![image](https://user-images.githubusercontent.com/16200219/129707669-7655bcc7-f8c0-44f9-86f6-eeefc7dc2a47.png)

Reply:
> However, this test is not running whenever the Health Check plugin is disabled though

This test will only be on the "Health Check" page if the health-check plugin is active. Here we are just adding one more callback in already [registered](https://plugins.trac.wordpress.org/browser/health-check/trunk/includes/class-health-check-site-status.php#L24) ajax call specifically for `amp_page_cache`.

Reply:
Right, but on the Site Health page the page cache test is not running because there is no action registered to return the test result.

Reply:
As of WordPress 5.2 and [r49154](https://core.trac.wordpress.org/changeset/49154), the Site Health tests can now be invoked via the REST API . I'll try reworking this to make use of that.

Reply:
Sorry, I mean as of WordPress 5.6: https://make.wordpress.org/core/2020/11/15/site-health-check-changes-in-5-6/

Reply:
Eww, but the Site Health plugin doesn't support the REST API async tests 😦

Reply:
OK, the issue is that the Health Check plugin is way out of date. The last release was 2 years ago. There are various issues reported in the GitHub repo about incompatibilities with current tests. So I'm going to add logic to prevent this REST-based async test from being added when on WP<5.6 and when this stale version of Health Check is active.